CONTRIBUTING.md

```diff
@@ -1,16 +1,51 @@
-# mongodb-operator
+# Contributing
 
-## Code overview
+## Overview
 
-TODO: expand on this section when spec doc is created
+This document explains the processes and practices recommended for contributing enhancements to this operator.
 
-## Intended use case
+- Generally, before developing enhancements to this charm, you should consider opening an issue explaining your use case.
+- If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev) or [Discourse](https://discourse.charmhub.io/).
+- Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library will help you a lot when working on new features or bug fixes.
+- All enhancements require review before being merged. Code review typically examines
+    - code quality
+    - test coverage
+    - user experience for Juju administrators of this charm.
+- Please help us out in ensuring easy to review branches by rebasing your pull request branch onto the `main` branch. This also avoids merge commits and creates a linear Git commit history.
 
-TODO: expand on this section when spec doc is created
 
-## Roadmap
 
-TODO: expand on this section when spec doc is created
+## Developing
+
+
+### Environment set up
+
+This operator charm can be deployed locally using [Juju on a localhost LXD cloud](https://juju.is/docs/olm/lxd). If you do not already have a Juju controller bootstrapped to a LXD `localhost` cloud, you can set one up by doing the following:
```

Comment:
> If you do not already have a Juju controller bootstrapped to a LXD `localhost` cloud

I'd probably omit the "LXD `localhost` cloud" part, as you can basically bootstrap a controller anywhere. If we change this we should also change the instructions down below.